x-pack/test/functional/services/aiops/explain_log_rate_spikes.ts

```diff
@@ -56,9 +58,93 @@ export function ExplainLogRateSpikesProvider({ getService }: FtrProviderContext)
       await testSubjects.existOrFail(`aiopsNoWindowParametersEmptyPrompt`);
     },
 
+    async assertNoResultsFoundEmptyPromptExist() {
+      await testSubjects.existOrFail(`aiopsNoResultsFoundEmptyPrompt`);
+    },
+
+    async clickDocumentCountChart() {
+      await elasticChart.waitForRenderComplete();
+      const el = await elasticChart.getCanvas();
+
+      await browser.getActions().move({ x: 0, y: 0, origin: el._webElement }).click().perform();
+    },
+
+    async clickRerunAnalysisButton(shouldRerun: boolean) {
+      await testSubjects.clickWhenNotDisabled(
+        `aiopsRerunAnalysisButton${shouldRerun ? ' shouldRerun' : ''}`
+      );
+
+      await retry.tryForTime(30 * 1000, async () => {
+        await testSubjects.existOrFail(
+          `aiopsRerunAnalysisButton${!shouldRerun ? ' shouldRerun' : ''}`
+        );
+      });
+    },
+
+    async assertAnalysisSectionExist() {
+      await retry.tryForTime(5000, async () => {
+        await testSubjects.existOrFail(`aiopsExplainLogRateSpikesAnalysis`);
+      });
+    },
+
+    async assertRerunAnalysisButton(shouldRerun: boolean) {
+      await testSubjects.existOrFail(
+        `aiopsRerunAnalysisButton${shouldRerun ? ' shouldRerun' : ''}`
+      );
+    },
+
+    async assertProgressTitle(expectedProgressTitle: string) {
+      await testSubjects.existOrFail('aiopProgressTitle');
+      const currentProgressTitle = await testSubjects.getVisibleText('aiopProgressTitle');
+      expect(currentProgressTitle).to.be(expectedProgressTitle);
+    },
+
     async navigateToIndexPatternSelection() {
       await testSubjects.click('mlMainTab explainLogRateSpikes');
       await testSubjects.existOrFail('mlPageSourceSelection');
     },
+
+    async getBrushSelectionWidth(selector: string) {
+      const brush = await testSubjects.find(selector);
+      const brushSelectionBefore = (await brush.findAllByClassName('selection'))[0];
+      const brushSelectionRectBefore = await brushSelectionBefore._webElement.getRect();
+      return brushSelectionRectBefore.width;
+    },
```

Comment:
Just a nit, but I think we don't need to the `Before` parts in the local variable names here as this method doesn't know about the overall test flow.

Reply:
Good catch, leftover from refactoring, fixed in [2d7291e](https://github.com/elastic/kibana/pull/138661/commits/2d7291e24e0219a94f18b557badae782eac26ed4).